librz/arch/isa/xtensa/xtensa_il.c

```diff
@@ -0,0 +1,2178 @@
+		{ FSR_I, 7, 1 },
+		{ FSR_U, 8, 1 },
+		{ FSR_O, 9, 1 },
+		{ FSR_Z, 10, 1 },
+		{ FSR_V, 11, 1 },
+		{ FSR_MBZ, 12, 20 },
+	},
+	.size = 7,
+	.width = 32,
+};
+
+static RzILOpPure *reg_field_set(const RegFieldTbl *tbl, ut32 field, RzILOpPure *orig, RzILOpPure *v) {
+	for (size_t i = 0; i < tbl->size; ++i) {
+		const RegField *f = tbl->tbl + i;
+		if (field == f->field) {
+			return tbl->width == 32
+				? DEPOSIT32(orig, U32(f->offset), U32(f->width), v)
+				: DEPOSIT64(orig, U32(f->offset), U32(f->width), v);
+		}
+	}
+	rz_warn_if_reached();
+	return NULL;
+}
+
+static RzILOpPure *reg_field_get(const RegFieldTbl *tbl, ut32 field, RzILOpPure *orig) {
+	for (size_t i = 0; i < tbl->size; ++i) {
+		const RegField *f = tbl->tbl + i;
+		if (field == f->field) {
+			return tbl->width == 32
+				? EXTRACT32(orig, U32(f->offset), U32(f->width))
+				: EXTRACT64(orig, U32(f->offset), U32(f->width));
+		}
+	}
+	rz_warn_if_reached();
+	return NULL;
+}
+
+#define PS_field_set(F, V) SETG("ps", reg_field_set(&ps_field_tbl, (F), VARG("ps"), (V)))
+#define PS_field_get(F)    reg_field_get(&ps_field_tbl, (F), VARG("ps"))
+#define PS_EXCM_CLEAR      PS_field_set(PS_EXCM, U32(0))
+
+#define FCR_field_set(F, V) SETG("fcr", reg_field_set(&fcr_field_tbl, (F), VARG("fcr"), (V)))
+#define FCR_field_get(F)    reg_field_get(&fcr_field_tbl, (F), VARG("fcr"))
+
+#define FSR_field_set(F, V) SETG("fsr", reg_field_set(&fsr_field_tbl, (F), VARG("fsr"), (V)))
+#define FSR_field_get(F)    reg_field_get(&fsr_field_tbl, (F), VARG("fsr"))
+
+static RzAnalysisLiftedILOp f_cons_(RzILOpEffect *x, RzILOpEffect *y) {
+	if (!(x && x->code == RZ_IL_OP_SEQ)) {
+		goto err;
+	}
+	RzILOpEffect *last = x;
+	while (last->op.seq.y && last->op.seq.y->code == RZ_IL_OP_SEQ) {
+		last = last->op.seq.y;
+	}
+	if (last->op.seq.y) {
+		RzILOpEffect *seq = RZ_NEW0(RzILOpEffect);
+		if (!seq) {
+			goto err;
+		}
+		seq->code = RZ_IL_OP_SEQ;
+		seq->op.seq.x = last->op.seq.y;
+		seq->op.seq.y = y;
+		last->op.seq.y = seq;
+	} else {
+		last->op.seq.y = y;
+	}
+	return x;
+err:
+	rz_warn_if_reached();
+	rz_il_op_effect_free(x);
+	rz_il_op_effect_free(y);
+	return NULL;
+}
+
+static RzILOpEffect *FSR_set(ut16 fs, RzILOpPure *v) {
+	RzILOpEffect *eff = SEQ2(
+		SETL("fsr_v", v),
+		NOP());
+	if (fs & FSR_V) {
+		eff = f_cons_(eff, FSR_field_set(FSR_V, B2U32(FEXCEPT(RZ_FLOAT_E_INVALID_OP, VARL("fsr_v")))));
+	}
+	if (fs & FSR_I) {
+		eff = f_cons_(eff, FSR_field_set(FSR_I, B2U32(FEXCEPT(RZ_FLOAT_E_INEXACT, VARL("fsr_v")))));
+	}
+	if (fs & FSR_O) {
+		eff = f_cons_(eff, FSR_field_set(FSR_O, B2U32(FEXCEPT(RZ_FLOAT_E_OVERFLOW, VARL("fsr_v")))));
+	}
+	if (fs & FSR_U) {
+		eff = f_cons_(eff, FSR_field_set(FSR_U, B2U32(FEXCEPT(RZ_FLOAT_E_UNDERFLOW, VARL("fsr_v")))));
+	}
+	if (fs & FSR_Z) {
+		eff = f_cons_(eff, FSR_field_set(FSR_Z, B2U32(FEXCEPT(RZ_FLOAT_E_DIV_ZERO, VARL("fsr_v")))));
+	}
+	rz_warn_if_fail(eff);
+	return eff;
+}
+
+static RzILOpEffect *ARi_set(RzILOpPure *i, RzILOpPure *v) {
```

Comment:
Is it possible to make it less nested somehow to not break formatting like that?